src/main/java/com/bbloggsbott/readytoserve/pages/service/DirectoryWatchingService.java

```diff
@@ -0,0 +1,119 @@
+package com.bbloggsbott.readytoserve.pages.service;
+
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+
+@Service
+public class DirectoryWatchingService {
+
+    @Autowired
+    private PagePathService pagePathService;
+
+    private List<Thread> watcherThreads = new ArrayList<>();
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @PreDestroy
+    public void destroy(){
+        for (Thread watcherThread: watcherThreads){
+            watcherThread.interrupt();
+        }
+    }
+
+    private void createDirectoryWatcher(String directory) throws IOException {
+        logger.info("Creating Directory watcher for {}", directory);
```

Comment:
The `createDirectoryWatcher` seems to be a bootstrapping function for creating the Watcher, correct ?

Reply:
Yup

Reply:
Not a good place for Service Class but will get refactored out eventually I guess

Reply:
I'll try pulling it into PageService as an AsyncMethod